consensus/hotstuff/README.md

```diff
@@ -141,49 +142,45 @@ When starting to look into the code, we suggest starting with `/consensus/hotstu
-* `/consensus/hotstuff/pacemaker` contains the implementation of Flow's basic PaceMaker, as described above.
-* `/consensus/hotstuff/persister` for performance reasons, the implementation maintains the consensus state largely in-memory. The `persister` stores the last entered view and the view of the latest voted block persistenlty on disk. This allows recovery after a crash without the risk of equivocation.       
-* `/consensus/hotstuff/runner` helper code for starting and shutting down the HotStuff logic safely in a multithreaded environment.  
-* `/consensus/hotstuff/validator` holds the logic for validating the HotStuff-relevant aspects of blocks, QCs, and votes
+* `/consensus/hotstuff/notifications`: All relevant events within the HotStuff logic are exported though a notification system. Notifications are used by _some_ HotStuff components internally to drive core logic (events from `VoteAggregator` and `TimeoutAggregator`), they also notify other components within the same node of relevant progress and are used for collecting HotStuff metrics.
+* `/consensus/hotstuff/pacemaker` contains the implementation of Flow's Active PaceMaker, as described above. Is responsible for protocol liveness. 
+* `/consensus/hotstuff/persister` for performance reasons, the implementation maintains the consensus state largely in-memory. The `persister` stores latest safety and liveness data persistently on disk. This allows recovery after a crash without the risk of equivocation.
+* `/consensus/hotstuff/safetyrules` tracks the latest vote, the latest timeout and determines whether to vote for a block and if it's safe to timeout current round.
+* `/consensus/hotstuff/signature` contains the implementation for threshold signature aggregation for all types of signatures that are used in HotStuff protocol. 
+* `/consensus/hotstuff/timeoutaggregator` caches timeouts on a per-view basis and orchestrates processing of timeouts and as result creation of TC.
+* `/consensus/hotstuff/timeoutcollector` performs timeout collection and processing, as part of processing validation and verification of timeouts. Builds TC if enough timeouts have been accumulated.
+* `/consensus/hotstuff/tracker` implements utility code for tracking the newest QC and TC in multithreaded environment.
+* `/consensus/hotstuff/validator` holds the logic for validating the HotStuff-relevant aspects of blocks, QCs, TC, and votes
 * `/consensus/hotstuff/verification` contains integration of Flow's cryptographic primitives (signing and signature verification) 
 * `/consensus/hotstuff/voteaggregator` caches votes on a per-block basis and builds a QC if enough votes have been accumulated.
-* `/consensus/hotstuff/voter` tracks the view of the latest vote and determines whether or not to vote for a block
+* `/consensus/hotstuff/votecollector` performs vote collection and processing, and builds a QC if enough votes have been accumulated.
```

Comment:
same here: I feel it would be more illustrative to start with `votecollector` and then disucss `voteaggregator`

Suggestion:
> * `/consensus/hotstuff/votecollector` encapsulates the logic for caching, validating, and aggregating votes for one particular view. Builds a QC if enough votes have been accumulated.
> * `/consensus/hotstuff/voteaggregator` orchestrates  the `VoteCollector`s for different views. It distributes votes to the respective  `VoteCollector`, notifies the `VoteCollector` about the arrival of their respective block, and prunes collectors that are no longer needed.